server/controllers/notification.js

```diff
@@ -0,0 +1,70 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const { senderEmail, receipientEmail, type, title, description} =
+    req.body;
+
+  const notification = await Notification.create({
+    type,
+    title,
+    description,
+    receipientEmail,
+    senderEmail,
+  });
+
+  res.status(201).json({
+    success: {
+      notification: {
+        id: notification._id,
+        type: notification.type,
+        title: notification.title,
+        description: notification.description,
+        read: notification.read,
+        receipientEmail: notification.receipientEmail,
+        senderEmail: notification.senderEmail,
+      },
+    },
+  });
+});
+
+exports.markNotificationAsRead = asyncHandler(async (req, res, next) => {
+  const { _id } = req.params;
+  const notification = await Notification.findById(_id);
+
+  if (!notification) {
+    res.status(404);
+    throw new Error("The notification does not exist");
+  }
+  notification.toggleReadStatus();
+
+  res.status(200).json({
+    success: {
+      notification: {
+        id: notification._id,
+        description: notification.description,
+        read: notification.read,
+      },
+    },
+  });
+});
+
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+  const notifications = await Notification.find();
+
+  if (!natifications) return res.json({ msg: "No results found" });
```

Comment:
```suggestion
  if (!notifications) return res.json({ msg: "No results found" });
```
Empty arrays are [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) so this won't execute when there aren't any notifications foudn. 

In fact, this will never execute since the `find` query won't ever return a `falsy` value - also note that [`await` will throw if/when the promise rejects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description) so `notifications` won't ever be `undefined` here (normally we handle this by wrapping `await`s in `try...catch` blocks but in this case `express-async-handler` package will provide this behaviour for us).

Lastly, it is arguably better to just return the empty result set and call it a `success` (see [this stackoverflow post](https://stackoverflow.com/questions/13366730/proper-rest-response-for-empty-table)).

Reply:
noted with thanks.